src/twisted/words/protocols/jabber/client.py

```diff
@@ -115,7 +115,7 @@ def initialize(self):
 
     def _cbAuthQuery(self, iq):
         jid = self.xmlstream.authenticator.jid
-        password = _coercedUnicode(self.xmlstream.authenticator.password)
```

Comment:
Reviewer note: assuming that the type assertion in `addElement` is restored this is not a behavioral change. This value is either passed to `addElement` or `hashPassword()`, which raises `TypeError` for non-`str` passwords.